dapp_runner/_util.py

```diff
@@ -2,30 +2,56 @@
 import socket
 from asyncio import Task
 from datetime import datetime, timezone
-from typing import Any
+from typing import Any, Generator
 
 import statemachine
 from colors import yellow
 
 from yapapi import Golem
 from yapapi import __version__ as yapapi_version
 
+from dapp_runner.metaclasses import SingletonMeta
 
-def get_free_port(range_start: int = 8080, range_end: int = 9090) -> int:
-    """Get the first available port on localhost within the specified range.
 
-    The range is inclusive on both sides (i.e. `range_end` will be included).
-    Raises `RuntimeError` when no free port could be found.
+class FreePortSingleton(metaclass=SingletonMeta):
```

Comment:
```suggestion
class FreePortGenerator(metaclass=SingletonMeta):
```

or 

```suggestion
class FreePortProvider(metaclass=SingletonMeta):
```


"Singleton" is just the pattern we use, not a functionality...

Reply:
>"Singleton" is just the pattern we use, not a functionality...

It is true but `FreePortProvider().get_free_port()` will not reflect in any way it is a singleton therefor might yield unexpected behavior from user stand point.

Anyway, I am ok with both approaches, will change to `FreePortProvider`